@mczernek/web browser improvements (#3691)
expo/expo: docs/pages/versions/unversioned/sdk/webbrowser.md

#### Arguments

- **url (_string_)** -- The url to open in the web browser.
- **arguments (_object_)** --
  Optional. A dictionaty with following key-value pairs:

>   - `toolbarColor` : a string describing a color of toolbar to be used by Custom Tab on Android. Required format `#AARRGGBB` or `#RRGGBB`. This is supported only on Android. Optional.

Reviewer: See for example this https://github.com/expo/expo/blob/8864b1cd7ce15f5f9e058360d6474e2d02d455fd/docs/pages/versions/unversioned/sdk/notifications.md#arguments-2 and follow convention in marking `optional` and `platform only` things 😄

>   - **enableBarCollapsing (_optional_) (_boolean_)** : flag determinig whether toolbar might be hiding when user scrolls website.

Reviewer 2: ```suggestion
  - **collapseToolbar (_optional_) (_boolean_)** : a boolean determining whether the toolbar should be hiding when a user scrolls the website
```

>   - **showTitle (_optional_) (_boolean_)** : _Android only_. Flag determining whether browser should show title of website on Toolbar.

Reviewer 2: ```suggestion
  - **showTitle (_optional_) (_boolean_)** : (_Android only_) a boolean determining whether the browser should show the title of website on the toolbar
```

>   - **package (_optional_) (_string_)** : _Android only_. Package name of a browser to be used to handle Custom Tabs. List of available packages is to be queried by [getCustomTabsSupportingBrowsers](#WebBrowser.getCustomTabsSupportingBrowsers) method.

Reviewer 2: ```suggestion
  - **browserPackage (_optional_) (_string_)** : (_Android only_) package name of the browser to be used when opening the URL. List of available packages is to be queried by [getCustomTabsSupportingBrowsers](#WebBrowser.getCustomTabsSupportingBrowsers) method
```

+ please run the docs server and check what will be the real anchor tag, it won't contain `.`, that's for sure

Author: Will change package to browserPackage in next PR if that's okay. I just want to minimize conflicts.

>   Note, that certain behavior depends on actual browser and its version. Some or all of arguments might be ignored.

Reviewer 2: ```suggestion
  Note that behavior customization options depend on the actual browser and its version. Some or all of the arguments may be ignored.
```

### `WebBrowser.openAuthSessionAsync(url, redirectUrl)`

> Opens the url with Safari in a modal on iOS using `SFAuthenticationSession`. The user will be asked whether to allow the app to authenticate using the given url. Unavailable on Android.

Reviewer 2: Was this always unavailable on Android and this is a docs update, or is this PR making this method unavailable on Android? 🤔

Author: It was always. There is no special Authentication mode in Custom Tabs.

The promise resolves with `{ type: 'dismiss' }`.

### `WebBrowser.getCustomTabsSupportingBrowsers`

> Available only on Android.

Reviewer: See here: https://github.com/expo/expo/blob/8864b1cd7ce15f5f9e058360d6474e2d02d455fd/docs/pages/versions/unversioned/sdk/notifications.md#notificationsdismissnotificationasynclocalnotificationid

> Returns two lists of applications package names supporting Custom Tabs. One is list of all applications and the other includes only default ones. However, this list might not be fully reliable, as they follow `PackageManager.getResolvingActivities` behavior with its flaws.

Author: I am not aware, just saying from my experience. I will look for it, however.

#### Returns

> The probime resolves with `{ packages: string[], default: string[] }`

Reviewer: `probime` - 🤔
